tests/test_user_defined_process_repository.py

```diff
@@ -114,3 +126,39 @@ def test_delete_miss(self, udp_db, zk):
         zk.delete.side_effect = NoNodeError()
         with pytest.raises(ProcessGraphNotFoundException):
             udp_db.delete(user_id="john", process_id="evi1")
+
+    @pytest.mark.parametrize(
+        ["zk_client_reuse", "expected_create_call_counts", "expected_cleanups"],
```

Comment:
I like that it tracks the evolution.

nit: `expected_create_call_counts` and `expected_cleanups`  are both tracking call counts so both parameters should be named similarly so perhaps have expected_cleanups als be expected_cleanup_call_counts

Reply:
done